src/client/MenuWindow.cpp

```diff
@@ -2,186 +2,372 @@
-        } 
-        else 
+        std::cerr << "Error while opening json ressources file /img/menu/menu.json" << std::endl;
+        exit(1);
+    }
+    std::string str((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+
+    std::unique_ptr<Json::CharReader> reader = std::unique_ptr<Json::CharReader>(Json::CharReaderBuilder().newCharReader());
+    Json::Value obj;
+    std::string errors;
+    reader->parse(str.c_str(), str.c_str() + str.size(), &obj, &errors);
+
+    loadText(obj["welcomeTexts"]);
+    loadButton(obj["welcomeButtons"]);
+
+    currentMenu = &newGameButtons;
+    currentText = &newGameTexts;
+    
+    newGameButtons.push_back(menuButtons[BUTTON_USERNAME]);
+    newGameButtons.push_back(menuButtons[BUTTON_SERVER]);
+    newGameButtons.push_back(menuButtons[BUTTON_PORT]);
+    
+    loadText(obj["newGameTexts"]);
+    loadButton(obj["newGameButtons"]);
+
+    currentMenu = &menuButtons;
+    currentText = &menuTexts;
+}
+
+void MenuWindow::loadText(Json::Value &data)
+{
+    for (auto &dataMenu : data)
+    {
+        currentText->emplace_back(dataMenu["text"].asString(), menuFont, dataMenu["size"].asInt());
+        if (dataMenu["bold"].asBool())
         {
-            return std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+            currentText->back().setStyle(sf::Text::Bold);
         }
+        currentText->back().setFillColor(sf::Color::Black);
+        currentText->back().setPosition((int)(gameEnginePtr->clientWindow->getSize().x 
+                                                - dataMenu["xOffset"].asFloat() 
+                                                * currentText->back().getLocalBounds().height 
+                                                - currentText->back().getLocalBounds().width), 
+
+                                        (int)(gameEnginePtr->clientWindow->getSize().y
+                                                - dataMenu["yOffset"].asFloat() 
+                                                * currentText->back().getLocalBounds().height));
     }
 }
+
+void MenuWindow::loadButton(Json::Value &data)
+{
+    for (auto &dataMenu : data)
+    {
+        currentMenu->emplace_back(  sf::Vector2f(dataMenu["width"].asFloat() 
+                                                    * menuTexts[0].getLocalBounds().width, 
+                                                 dataMenu["height"].asFloat() 
+                                                    * menuTexts[0].getLocalBounds().height), 
+                                                    
+                                    sf::Vector2f(   menuTexts[0].getPosition().x 
+                                                    + dataMenu["x"].asFloat() 
+                                                    * menuTexts[0].getLocalBounds().width, 
+                                                    menuTexts[0].getPosition().y 
```

Comment:
Make your function work also for that (add parameter or something)

Reply:
Done